datasetinsights/io/compression.py

```diff
@@ -0,0 +1,99 @@
+import gzip
+import logging
+import os
+import tarfile
+import zipfile
+
+import filetype
+
+logger = logging.getLogger(__name__)
+
+
+class ZipFileCompression:
```

Comment:
After looking around abit i found this. 

import shutil
This function could replace the different compressors.
shutil.unpack_archive()

Reply:
Okay, I'll implement this.

Reply:
Remember to pull i just pushed some tests maybe they are not needed at this point, if this is implemented. Feel free to remove them if need be.

Reply:
After reading the documentation, which states: `format is the archive format: one of “zip”, “tar”, “gztar”, “bztar”, or “xztar”`. That is they only support these 5 formats rest of the formats we'll have to register. Also, out of these 5 only 2 are most popular which means apart from these 2 we'll have to anyway implement other formats. Libraries like pytorch as well have their own [implementation](https://github.com/pytorch/vision/blob/dcfcc8675393749e852d8956b580b1660dfbea2d/torchvision/datasets/utils.py#L214).